File: programming-concepts/random-numbers/java/random_number.java
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science,
part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.util.Random;

class RandomNumbers
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args)
    {
        Random rnd = new Random();
        int random_number = rnd.nextInt(1, 101);
        System.out.println(random_number);
    }


}

File: programming-concepts/random-numbers/python/random_number.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0

from random import randint

def main():
    random_number = randint(1,100)
    print(random_number)

    
# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()